src/main/java/seedu/address/model/util/Date.java

```
@@ -0,0 +1,108 @@
+package seedu.address.model.util;
+
+import static java.util.Objects.requireNonNull;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+import seedu.address.logic.parser.DateKeyword;
+
+/**
+ * Represents Dates that are used in MyInsuRec
+ */
+public abstract class Date implements Comparable<Date> {
+    public final LocalDate date;
+
+    /**
+     * Constructs a {@code Date}.
+     *
+     * @param date A valid date.
+     */
+    public Date(LocalDate date) {
+        requireNonNull(date);
+        this.date = date;
+    }
+
+    private boolean isAfterDate(LocalDate other) {
+        return date.compareTo(other) >= 0;
+    }
+
+    private boolean isBeforeDate(LocalDate other) {
+        return date.compareTo(other) <= 0;
+    }
+
+    /**
+     * Returns true is and only if the date falls within the period {@code keyword}
+     * with respect to a givenDate {@code givenDate}.
+     */
+    public boolean isInPeriod(DateKeyword keyword, LocalDate givenDate) {
+        LocalDate startDate;
+        LocalDate endDate;
+
+        switch (keyword) {
+        case ALL_TIME:
+            return true;
+        case TOMORROW:
+            startDate = givenDate.plusDays(1);
+            endDate = givenDate.plusDays(1);
+            break;
+        case THIS_MONTH:
+            startDate = givenDate.withDayOfMonth(1);
+            endDate = givenDate.withDayOfMonth(givenDate.getMonth().length(givenDate.isLeapYear()));
+            break;
+        case THIS_WEEK:
+            startDate = givenDate;
+            endDate = givenDate.plusDays(7);
+            break;
+        default:
+            startDate = null;
+            endDate = null;
+        }
+        requireNonNull(startDate);
+        requireNonNull(endDate);
```

I guess for default you can just throw an exception directly?